gentoopm/basepm/upstream.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/python
+# 	vim:fileencoding=utf-8
+# (c) 2024 Anna <cyber@sysrq.in>
+# Released under the terms of the 2-clause BSD license.
+
+from dataclasses import dataclass
+from enum import Enum, auto
+
+
+class PMUpstreamMaintainerStatus(Enum):
+    """ Maintainer status enumeration. """
+
+    NONE = auto()
+    ACTIVE = auto()
+    INACTIVE = auto()
+
+
+@dataclass(frozen=True, order=True)
+class PMUpstreamMaintainer:
+    """ Representation of an upstream maintainer"""
+
+    name: str
+    email: str | None = None
+    status: PMUpstreamMaintainerStatus = PMUpstreamMaintainerStatus.NONE
+
+    def __str__(self) -> str:
+        if self.name and self.email:
+            return f"{self.name} <{self.email}>"
+        return self.name
+
+
+@dataclass(frozen=True)
+class PMUpstreamRemoteID:
+    """
+    Representation of a remote-id.
+    """
+
+    attr: str
```

Comment:
Use a more meaningful name.

Reply:
I'll switch to what pkgcraft uses (`name` and `site`)